openedx/features/course_experience/tests/views/test_masquerade.py

```diff
@@ -0,0 +1,128 @@
+"""
+Tests for masquerading functionality on course_experience
+"""
+
+
+import json
+
+import six
+from django.urls import reverse
+
+from lms.djangoapps.commerce.models import CommerceConfiguration
+from openedx.core.djangoapps.waffle_utils.testutils import override_waffle_flag
+from openedx.features.course_experience import DISPLAY_COURSE_SOCK_FLAG, SHOW_UPGRADE_MSG_ON_COURSE_HOME
+from student.roles import CourseStaffRole
+from student.tests.factories import CourseEnrollmentFactory, UserFactory
+from xmodule.modulestore.tests.django_utils import SharedModuleStoreTestCase
+from xmodule.modulestore.tests.factories import CourseFactory
+from xmodule.partitions.partitions import ENROLLMENT_TRACK_PARTITION_ID
+from xmodule.partitions.partitions_service import PartitionService
+
+from .helpers import add_course_mode
+from .test_course_home import TEST_UPDATE_MESSAGE, course_home_url
+from .test_course_sock import TEST_VERIFICATION_SOCK_LOCATOR
+
+TEST_PASSWORD = 'test'
+UPGRADE_MESSAGE_CONTAINER = 'section-upgrade'
+
+
+class MasqueradeTestBase(SharedModuleStoreTestCase):
+    """
+    Base test class for masquerading functionality on course_experience
+    """
+    @classmethod
+    def setUpClass(cls):
+        super(MasqueradeTestBase, cls).setUpClass()
+
+        # Create four courses
+        cls.verified_course = CourseFactory.create()
+        cls.masters_course = CourseFactory.create()
+        # Assign each verifiable course an upgrade deadline
+        add_course_mode(cls.verified_course, upgrade_deadline_expired=False)
+        add_course_mode(cls.masters_course, upgrade_deadline_expired=False)
+        add_course_mode(cls.masters_course, mode_slug='masters', mode_display_name='Masters')
+
+    def setUp(self):
+        super(MasqueradeTestBase, self).setUp()
+        self.course_staff = UserFactory.create()
+        CourseStaffRole(self.verified_course.id).add_users(self.course_staff)
+        CourseStaffRole(self.masters_course.id).add_users(self.course_staff)
+
+        # Enroll the user in the four courses
+        CourseEnrollmentFactory.create(user=self.course_staff, course_id=self.verified_course.id)
+        CourseEnrollmentFactory.create(user=self.course_staff, course_id=self.masters_course.id)
+
+        CommerceConfiguration.objects.create(enabled=True, checkout_on_ecommerce_service=True)
```

Comment:
What does this do? How does this differ from ecommerce? Why do we need it set up in order to test masquerading?

Reply:
An old relic that's not necessary now. Thank you for spotting it.